### ISSUE TITLE
debugger: coreDumpPath ignored on mac, even when remote debugging via pipeTransport

### ISSUE THREAD

**Issue by joelhock** [microsoft/vscode-cpptools#840]
I use vscode on a mac to interact with gdb running on linux using pipeTransport.  Everything works great, except when trying to look at a core file.  I have something like:
```json
            "coreDumpPath": "/usr/scratch/theprogram.core",

            "request": "launch",
            "type": "cppdbg",
            "program": "theprogram",

            "MIMode": "gdb",
            "logging": {
                "trace": true,
                "traceResponse": true,
                "engineLogging": true
            },
```

So, `coreDumpPath` is specified, but the logging reports `IsCoreDump: false`:

```json
{
    "category": "telemetry",
    "output": "VS/Diagnostics/Debugger/Launch",
    "data": {
        "VS.Diagnostics.Debugger.ImplementationName": "Microsoft.MIDebugEngine",
        "VS.Diagnostics.Debugger.EngineVersion": "14.0.40526.1",
        "VS.Diagnostics.Debugger.HostVersion": "1.0.30525.1",
        "VS.Diagnostics.Debugger.AdapterId": "cppdbg",
        "VS.Diagnostics.Debugger.Launch.Duration": 10366,
        "VS.Diagnostics.Debugger.Launch.IsCoreDump": false,
        "VS.Diagnostics.Debugger.VisualizerFileUsed": false,
        "VS.Diagnostics.Debugger.SourceFileMappings": 0
    },
    "type": "output"
}
```

I'm not sure if the issue stems from the version of the debugging support that vscode downloads for the mac being too old, or whether the fact that macs can't debug corefiles natively is somehow preventing the option from being respected, even though the debugger is running on a remote system that does support it.

**Comment by pieandcakes** [comment]
@joelhock I've never tried on mac before. We don't download gdb to the mac so if the version you have is older that might be an issue. I also don't know if pipeTransport and core dump debugging are compatible. 

Can you try and enable `"logging": { "engineLogging": true }` and when you start debugging you will see a lot of lines at the top that say something like `LocalLaunchOptions` or `PipeLaunchOptions` and post that block?

**Comment by joelhock** [comment]
You're misunderstanding a bit.  gdb isn't running on the mac at all;  I'm using pipeTransport to run gdb on a linux box and the gdb mi traffic is being piped between vscode on the mac and gdb running on the linux box.

I don't quite understand how pipeTransport fits into the whole process, but I would think that loading the corefile would be part of the gdb mi commands that vscode sends, so I would imagine there is some problem outside of the pipe transport, notably since some piece on the mac is outputting the line `"VS.Diagnostics.Debugger.Launch.IsCoreDump": false,`, but maybe that someone needs to get into the PipeLaunchOptions.

If you still think it's an issue, here are the PipeLaunchOptions (I was already running with `engineLogging` enabled.)
```
1: (218) LaunchOptions<PipeLaunchOptions xmlns='http://schemas.microsoft.com/vstudio/MDDDebuggerOptions/2014'
1: (227) LaunchOptions  ExePath='linux.debug/bin/dview'
1: (227) LaunchOptions  WorkingDirectory='/scratch/joel/remote/dev'
1: (227) LaunchOptions  ExeArguments=''
1: (227) LaunchOptions MIMode='gdb'
1: (227) LaunchOptions  PipePath='/usr/local/hrt-localcoding/bin/vscode_gdb_remoter_and_path_munger.sh'
1: (227) LaunchOptions  PipeArguments='/Volumes/hrt/dev  "gdb --interpreter=mi"'
1: (227) LaunchOptions  PipeCwd='/Volumes/dev'
1: (227) LaunchOptions>
1: (227) LaunchOptions    <SetupCommands>
1: (228) LaunchOptions        <Command IgnoreFailures='false' Description='Enable pretty-printing for gdb'>-enable-pretty-printing</Command>
1: (228) LaunchOptions    </SetupCommands>
1: (228) LaunchOptions</PipeLaunchOptions>
```
The framework assemblies that vscode downloads at https://go.microsoft.com/fwlink/?LinkId=816539 are from 06-01-2016, so it's possible things have been fixed since then?  I tried rolling my own assemblies according to the directions in
https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/Debugger/How%20To%20Debug%20MIEngine.md
but I never got them to work on my mac, even with a newer version of mono (seemed like it was picking up some parts of the old install, maybe from the global cache--I'm not too familiar with mono on osx.)

here's a more complete output of things after the PipeLaunchOptions:
```
1: (341) Starting: "/usr/local/hrt-localcoding/bin/vscode_gdb_remoter_and_path_munger.sh" /Volumes/hrt/dev  "gdb --interpreter=mi"
1: (878) ->=thread-group-added,id="i1"
1: (879) ->~"GNU gdb (Debian 7.11.1-2~bpo8+1) 7.11.1\n"
1: (879) ->~"Copyright (C) 2016 Free Software Foundation, Inc.\n"
1: (880) ->~"License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.  Type \"show copying\"\nand \"show warranty\" for details.\n"
1: (880) ->~"This GDB was configured as \"x86_64-linux-gnu\".\nType \"show configuration\" for configuration details."
1: (880) ->~"\nFor bug reporting instructions, please see:\n"
1: (880) ->~"<http://www.gnu.org/software/gdb/bugs/>.\n"
1: (880) ->~"Find the GDB manual and other documentation resources online at:\n<http://www.gnu.org/software/gdb/documentation/>.\n"
1: (880) ->~"For help, type \"help\".\n"
1: (880) ->~"Type \"apropos word\" to search for commands related to \"word\".\n"
1: (881) ->~"Sourcing /atl/make/hrt.gdbinit\n"
1: (1002) ->=cmd-param-changed,param="auto-load safe-path",value="/"
1: (1002) ->=cmd-param-changed,param="debug-file-directory",value="/build/linux/cache:/usr/lib/debug"
1: (1002) ->=cmd-param-changed,param="history filename",value="/home/joel/.gdb_history"
1: (1002) ->=cmd-param-changed,param="history save",value="on"
1: (1002) ->(gdb)
1: (1005) <-1001-gdb-set target-async on
1: (1068) ->1001^done
1: (1068) ->(gdb)
1: (1069) 1001: elapsed time 64
1: (1076) <-1002-enable-pretty-printing
1: (1159) ->1002^done
1: (1160) ->(gdb)
1: (1160) 1002: elapsed time 83
1: (1160) <-1003-interpreter-exec console "set pagination off"
1: (1215) ->=cmd-param-changed,param="pagination",value="off"
1: (1215) ->1003^done
1: (1215) ->(gdb)
1: (1215) 1003: elapsed time 55
1: (1215) <-1004-gdb-set solib-search-path linux.debug/bin:
1: (1271) ->1004^done
1: (1272) ->(gdb)
1: (1272) 1004: elapsed time 56
1: (1272) <-1005-gdb-set stop-on-solib-events 1
1: (1336) ->1005^done
1: (1336) ->(gdb)
1: (1336) 1005: elapsed time 64
1: (1336) <-1006-environment-cd /scratch/joel/remote/dev
1: (1399) ->1006^done
1: (1399) ->(gdb)
1: (1399) 1006: elapsed time 63
1: (1400) <-1007-file-exec-and-symbols linux.debug/bin/dview
1: (7223) ->1007^done,value="/scratchq/infra/joel/remote/dev/linux.debug/build:$cdir:$cwd"
1: (7223) ->(gdb)
1: (7225) 1007: elapsed time 5825
1: (7227) <-1008-interpreter-exec console "show architecture"
1: (7333) ->~"The target architecture is set automatically (currently i386:x86-64)\n"
1: (7333) ->1008^done
1: (7333) 1008: elapsed time 106
1: (7334) ->(gdb)
1: (7339) <-1009-break-insert main
1: (7434) ->1009^done,bkpt={number="1",type="breakpoint",disp="keep",enabled="y",addr="0x000000000041c524",func="main(int, char**)",file="atl/src/tools/dview.cc",fullname="/Volumes/hrt/dev/atl/src/tools/dview.cc",line="190",thread-groups=["i1"],times="0",original-location="main"}
1: (7435) ->(gdb)
1: (7437) 1009: elapsed time 98
E output: {"category":"telemetry","output":"VS/Diagnostics/Debugger/Launch","data":{"VS.Diagnostics.Debugger.ImplementationName":"Microsoft.MIDebugEngine","VS.Diagnostics.Debugger.EngineVersion":"14.0.40526.1","VS.Diagnostics.Debugger.HostVersion":"1.0.30525.1","VS.Diagnostics.Debugger.AdapterId":"cppdbg","VS.Diagnostics.Debugger.Launch.Duration":7353,"VS.Diagnostics.Debugger.Launch.IsCoreDump":false,"VS.Diagnostics.Debugger.VisualizerFileUsed":false,"VS.Diagnostics.Debugger.SourceFileMappings":0},"type":"output"}
 R: {"success":true,"message":null,"request_seq":2,"command":"launch","body":null,"running":false,"refs":null,"seq":0,"type":"response"}
E initialized: {"type":"initialized"}
C setFunctionBreakpoints: {"breakpoints":[]}
 R: {"success":true,"message":null,"request_seq":3,"command":"setFunctionBreakpoints","body":{"breakpoints":[]},"running":false,"refs":null,"seq":0,"type":"response"}
C configurationDone: null
 R: {"success":true,"message":null,"request_seq":4,"command":"configurationDone","body":null,"running":false,"refs":null,"seq":0,"type":"response"}
C threads: null
E output: {"category":"stdout","output":"=thread-group-added,id=\"i1\"\nGNU gdb (Debian 7.11.1-2~bpo8+1) 7.11.1\nCopyright (C) 2016 Free Software Foundation, Inc.\nLicense GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.  Type \"show copying\"\nand \"show warranty\" for details.\nThis GDB was configured as \"x86_64-linux-gnu\".\nType \"show configuration\" for configuration details.\nFor bug reporting instructions, please see:\n<http://www.gnu.org/software/gdb/bugs/>.\nFind the GDB manual and other documentation resources online at:\n<http://www.gnu.org/software/gdb/documentation/>.\nFor help, type \"help\".\nType \"apropos word\" to search for commands related to \"word\".\nSourcing /atl/make/hrt.gdbinit\n=cmd-param-changed,param=\"auto-load safe-path\",value=\"/\"\n=cmd-param-changed,param=\"debug-file-directory\",value=\"/build/linux/cache:/usr/lib/debug\"\n=cmd-param-changed,param=\"history filename\",value=\"/home/joel/.gdb_history\"\n=cmd-param-changed,param=\"history save\",value=\"on\"\n=cmd-param-changed,param=\"pagination\",value=\"off\"\n","data":null,"type":"output"}
=thread-group-added,id="i1"
GNU gdb (Debian 7.11.1-2~bpo8+1) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
Sourcing /atl/make/hrt.gdbinit
=cmd-param-changed,param="auto-load safe-path",value="/"
=cmd-param-changed,param="debug-file-directory",value="/build/linux/cache:/usr/lib/debug"
=cmd-param-changed,param="history filename",value="/home/joel/.gdb_history"
=cmd-param-changed,param="history save",value="on"
=cmd-param-changed,param="pagination",value="off"
 R: {"success":true,"message":null,"request_seq":5,"command":"threads","body":{"threads":[]},"running":false,"refs":null,"seq":0,"type":"response"}
1: (7503) <-1010-exec-run
1: (7570) ->=thread-group-started,id="i1",pid="24364"
1: (7570) ->=thread-created,id="1",group-id="i1"
1: (7575) <-1011-thread-info 1
1: (7608) ->=library-loaded,id="/lib64/ld-linux-x86-64.so.2",target-name="/lib64/ld-linux-x86-64.so.2",host-name="/lib64/ld-linux-x86-64.so.2",symbols-loaded="0",thread-group="i1"
1: (7650) ->1010^running
1: (7650) ->*running,thread-id="all"
1: (7650) 1010: elapsed time 147
1: (7650) ->(gdb)
1: (7650) ->1011^done,threads=[{id="1",target-id="process 24364",name="dview",state="running",core="9"}]
1: (7650) ->(gdb)
1: (7652) 1011: elapsed time 76
1: (7658) ->~"Stopped due to shared library event (no libraries added or removed)\n"
E thread: {"reason":"started","threadId":24364,"type":"thread"}
1: (7658) ->*stopped,reason="solib-event",thread-id="1",stopped-threads="all",core="9"
E output: {"category":"stdout","output":"Stopped due to shared library event (no libraries added or removed)\n","data":null,"type":"output"}
Stopped due to shared library event (no libraries added or removed)
C threads: null
 R: {"success":true,"message":null,"request_seq":6,"command":"threads","body":{"threads":[{"id":24364,"name":"dview"}]},"running":false,"refs":null,"seq":0,"type":"response"}
1: (7667) <-1012-interpreter-exec console "info sharedlibrary"
1: (7756) ->~"From                To                  Syms Read   Shared Object Library\n"
1: (7756) ->~"0x00007ffff7ddcae0  0x00007ffff7df5140  Yes         /lib64/ld-linux-x86-64.so.2\n"
1: (7756) ->1012^done
1: (7756) ->(gdb)
1: (7756) 1012: elapsed time 89
E output: {"category":"console","output":"Loaded '/lib64/ld-linux-x86-64.so.2'. Symbols loaded.\n","data":null,"type":"output"}
Loaded '/lib64/ld-linux-x86-64.so.2'. Symbols loaded.
1: (7760) <--exec-continue
1: (7817) ->^running
1: (7817) ->*running,thread-id="all"
1: (7818) ->(gdb)
1: (7830) ->=library-loaded,id="/Volumes/hrt/system-includes/usr/lib/x86_64-linux-gnu/libmysqlclient.so.18",target-name="/Volumes/hrt/system-includes/usr/lib/x86_64-linux-gnu/libmysqlclient.so.18",host-name="/Volumes/hrt/system-includes/usr/lib/x86_64-linux-gnu/libmysqlclient.so.18",symbols-loaded="0",thread-group="i1"

...

1: (8006) ->~"[Thread debugging using libthread_db enabled]\n"
1: (8006) ->~"Using host libthread_db library \"/lib/x86_64-linux-gnu/libthread_db.so.1\".\n"
E output: {"category":"stdout","output":"[Thread debugging using libthread_db enabled]\n","data":null,"type":"output"}
[Thread debugging using libthread_db enabled]
E output: {"category":"stdout","output":"Using host libthread_db library \"/lib/x86_64-linux-gnu/libthread_db.so.1\".\n","data":null,"type":"output"}
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
1: (8185) ->~"Stopped due to shared library event:\n"
1: (8185) ->~"  Inferior loaded /usr/lib/libhugetlbfs.so\n"
E output: {"category":"stdout","output":"Stopped due to shared library event:\n","data":null,"type":"output"}
Stopped due to shared library event:
E output: {"category":"stdout","output":"    /usr/lib/x86_64-linux-gnu/libmysqlclient.so.18\n","data":null,"type":"output"}

...

1: (8286) <-1013-interpreter-exec console "info sharedlibrary"
1: (8415) ->~"From                To                  Syms Read   Shared Object Library\n"
1: (8416) ->~"0x00007ffff749f360  0x00007ffff750a1b6  Yes         /usr/lib/x86_64-linux-gnu/libmysqlclient.so.18\n"

...

1: (8516) ->1013^done
1: (8516) ->(gdb)
1: (8516) 1013: elapsed time 229
E output: {"category":"console","output":"Loaded '/usr/lib/x86_64-linux-gnu/libmysqlclient.so.18'. Symbols loaded.\n","data":null,"type":"output"}
Loaded '/usr/lib/x86_64-linux-gnu/libmysqlclient.so.18'. Symbols loaded.

...

1: (8521) <--exec-continue
1: (8582) ->^running
1: (8582) ->*running,thread-id="all"

...

The program 'linux.debug/bin/dview' has exited with code 1 (0x00000001).

E exited: {"exitCode":1,"type":"exited"}
E terminated: {"type":"terminated"}
E output: {"category":"telemetry","output":"VS/Diagnostics/Debugger/DebugCompleted","data":{"VS.Diagnostics.Debugger.ImplementationName":"Microsoft.MIDebugEngine","VS.Diagnostics.Debugger.EngineVersion":"14.0.40526.1","VS.Diagnostics.Debugger.HostVersion":"1.0.30525.1","VS.Diagnostics.Debugger.AdapterId":"cppdbg","VS.Diagnostics.Debugger.DebugCompleted.BreakCounter":0},"type":"output"}
1: (8846) <--gdb-exit
C disconnect: {"restart":false}
 R: {"success":true,"message":null,"request_seq":7,"command":"disconnect","body":null,"running":false,"refs":null,"seq":0,"type":"response"}
```

**Comment by pieandcakes** [comment]
@joelhock A couple of things:

>The framework assemblies that vscode downloads at https://go.microsoft.com/fwlink/?LinkId=816539 are from 06-01-2016, so it's possible things have been fixed since then?

These are the mono framework assemblies and doesn't affect how the debugger works. They are used to run the intermediate layer as a framework such as .NET Framework. 

It looks like in the pipeTransport case we might not be forwarding the coredump path to MIEngine. I'll track this as a bug and investigate further when I have some time.

**Comment by joelhock** [comment]
ok, that makes sense.  thanks!

**Comment by pieandcakes** [comment]
@joelhock This should be fixed in `0.12.1` which was released yesterday. Can you see if this is still an issue?

**Comment by joelhock** [comment]
@pieandcakes 
sorry, not working yet.  I do see where we now have:
`1: (241) LaunchOptions CoreDumpPath='/usr/scratch/core/dview.1012.core'`
which was not present before, so that's good.

however, the program still runs instead of having a corefile loaded, and the line
`E output: {"category":"telemetry","output":"VS/Diagnostics/Debugger/Launch","data":{"VS.Diagnostics.Debugger.ImplementationName":"Microsoft.MIDebugEngine","VS.Diagnostics.Debugger.EngineVersion":"14.0.40717.1","VS.Diagnostics.Debugger.HostVersion":"1.0.30717.1","VS.Diagnostics.Debugger.AdapterId":"cppdbg","VS.Diagnostics.Debugger.Launch.Duration":9182,"VS.Diagnostics.Debugger.Launch.IsCoreDump":false,"VS.Diagnostics.Debugger.VisualizerFileUsed":false,"VS.Diagnostics.Debugger.SourceFileMappings":0},"type":"output"}
`
is present with `"VS.Diagnostics.Debugger.Launch.IsCoreDump":false,"` , so the `CoreDumpPath` in `LaunchOptions` isn't actually doing what it should.

- VSCode Version: Code 1.14.1 (2648980a697a4c8fb5777dcfb2ab110cec8a2f58, 2017-07-13T19:08:57.197Z)
- OS Version: Darwin x64 16.5.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
project-manager|ale|0.18.1
githistory|don|0.2.1
python|don|0.6.7
gitlens|eam|4.3.1
vscode-great-icons|emm|2.1.11
cpptools|ms-|0.12.1
rewrap|stk|1.4.1
better-align|wwm|1.1.6
